packages/zowe-explorer/src/shared/utils.ts

```diff
@@ -85,19 +85,21 @@ export function labelRefresh(node: vscode.TreeItem): void {
  * @param {sessNode} IZoweTreeNode
  * @param {profile} IProfileLoaded
  *************************************************************************************************************/
-export function refreshTree(sessNode: IZoweTreeNode) {
-    const allProf = Profiles.getInstance().getProfiles();
+export async function refreshTree(sessNode: IZoweTreeNode) {
+    const allProf = await Profiles.getInstance().getProfiles();
+    const baseProf = await Profiles.getInstance().getBaseProfile();
     for (const profNode of allProf) {
         if (sessNode.getProfileName() === profNode.name) {
             sessNode.getProfile().profile = profNode.profile;
-            const SessionProfile = profNode.profile as ISession;
-            if (sessNode.getSession().ISession !== SessionProfile) {
-                sessNode.getSession().ISession.user = SessionProfile.user;
-                sessNode.getSession().ISession.password = SessionProfile.password;
-                sessNode.getSession().ISession.base64EncodedAuth = SessionProfile.base64EncodedAuth;
-                sessNode.getSession().ISession.hostname = SessionProfile.hostname;
-                sessNode.getSession().ISession.port = SessionProfile.port;
-                sessNode.getSession().ISession.rejectUnauthorized = SessionProfile.rejectUnauthorized;
+            const combinedSessionProfile = (await Profiles.getInstance().getCombinedProfile(profNode, baseProf))
+                .profile;
+            const sessionNode = sessNode.getSession();
+            for (const prop of Object.keys(combinedSessionProfile)) {
+                if (prop === "host") {
+                    sessionNode.ISession.hostname = combinedSessionProfile[prop];
```

Comment:
This goes back to https://github.com/zowe/zowe-cli/issues/855#issuecomment-724778568
I really wish we didn't have to do this, but this is coming out of Imperative.
@dkelosky, thoughts on including this in `Imperative@next` and `CLI@next`?

Reply:
That sounds good.  Do you want `hostname` gone entirely or both to exist, e.g. make `hostname` take precedence over `host` if it's there?

Reply:
Personally, I don't mind either way, but we (in ZE) mostly use `host` since that's what profiles contain.

However, checkout this out:
CLI:
![image](https://user-images.githubusercontent.com/37381190/99696560-d3944700-2a5c-11eb-8076-be088acb5785.png)

Imperative:
![image](https://user-images.githubusercontent.com/37381190/99696781-135b2e80-2a5d-11eb-98f0-0c06725f7b9f.png)

🤣

Reply:
if we could be backwards compatible - I think that would be the preferred route to take

Reply:
@dkelosky @zFernand0 The reason for this is because the APIs are using IProfileLoaded/IProfile format when a request is made. But when creating a session, the response that we get is in Session/ISession format. So we need to transfer the ISession to IProfile so that we could fire a request. This transfer is not a problem because they are the same except for the hostname.